realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/QueryBasedSyncTests.java

```diff
@@ -336,6 +335,69 @@ public void downloadLimitedData() throws InterruptedException {
         });
     }
 
+    @Test
+    @RunTestInLooperThread
+    public void initialDataAndWaitForRemoteInitialData() throws InterruptedException {
+        SyncUser user = UserFactory.createUniqueUser(Constants.AUTH_URL);
+        createServerData(user, Constants.SYNC_SERVER_URL);
+
+        // Create partial Realm that will wait for the subscriptions
+        final SyncConfiguration partialSyncConfig = configurationFactory.createSyncConfigurationBuilder(user, Constants.SYNC_SERVER_URL)
+                .name("partialSync")
+                .initialData(r -> {
+                    r.where(PartialSyncObjectA.class).greaterThan("number", 5).subscribe("my-sub");
+                })
+                .waitForInitialRemoteData()
+                .modules(new PartialSyncModule())
+                .build();
+        Realm realm = Realm.getInstance(partialSyncConfig);
+        looperThread.closeAfterTest(realm);
+
+        // Check the state of subscriptions
+        assertEquals(6, realm.getSubscriptions().size());
```

Comment:
We have 1 subscription in `initialData` where does the 6 come from?

Reply:
Sync creates them automatically for the permission classes. I'll add a note a about it.